packages/hierarchy-browser/src/hooks/useControls.ts

```diff
@@ -0,0 +1,29 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import { useMemo } from 'react'
+import { IControls } from '../types'
+
+export function useControls(
+	controls?: IControls,
+): [boolean, boolean, boolean, boolean] {
+	return useMemo(() => {
+		let level = true
+		let membership = true
+		let exportData = true
+		let filterData = true
+		if (controls) {
+			level = controls.showLevel !== undefined ? controls.showLevel : level
+			membership =
+				controls.showMembership !== undefined
+					? controls.showMembership
+					: membership
+			filterData =
+				controls.showFilter !== undefined ? controls.showFilter : filterData
```

Comment:
The safest check here is `x != null`, which accounts for both null and undefined